### ISSUE TITLE
Add enforceEx(), is it possible to get which parameter was matched in enforcer.enforce()?

### ISSUE THREAD

**Issue by ankitaShrma** [casbin/node-casbin#282]
I have a policy:
```
p, role, group, path, method
```
roles are added:
```
g, user, role, domain
g2, domain, group
```
where domain = id

I am using node-casbin for authorization using policies like above. The result from **enforcer.enforce()** works well to check if the api is authorized or not.
My issue is then to get all the permissions for given user in the given domain based on the allowed enforcer matcher.
For eg: `r = user:abc, /data/123, put`
```
r = user, path, method

p, role:admin, group:api, /data/:id, get|put|delete
p, role:admin, group:app, button, show|edit

g, user:abc, role:admin, domain:123
g2, domain:123, group:api

m =(g(r.user, p.role, "domain:"+keyGet2Custom(r.path, p.path, ":id")) && g2("domain:"+keyGet2Custom(r.path, p.path, ":id"), p.domain) && keyMatch2(r.path, p.path) && regexMatch(r.method, p.method)) 
```
matcher will give true for /data/123

Now I want to get all the permission for the given user, for which I need to know which policy was matched so that I can know the domain:123 from the given request path i.e I need to know what the data is in `id` in path.
I did check the code and found that what I need to make it work was the `parameters` variable in **privateEnforcer**.
Is there a way to get such parameter which was matched?

**Comment by hsluoyz** [comment]
@ankitaShrma see: https://casbin.org/docs/en/data-permissions

**Comment by ankitaShrma** [comment]
Rather than getting all permissions, I actually needed to know the exact policy that returned true from the matcher in enforcer.
The thing is I do not know the `domain` for the given user to get the permissions from implicit functions.
So like in above example:
```
enforcer.enforce('user:abc', '/data/123', 'put') // gives true with my matcher 
```
The above gives true because my matcher checks from:
```
g, user:abc, role:owner, domain:123
g2, domain:123, group:main
g2, group:main, group:api
p, role:owner, group:api, /data/:id, put
p, role:owner, group:api, /data, put
```
using custom `keyGet2Custom` function (since it is not in node version): I can check in matcher:
```
m = g(r.user, p.role, "domain:"+keyGet2Custom(r.path, p.path, ":id")) && g2("domain:"+keyGet2Custom(r.path, p.path, ":id"), p.domain)
```
This will match true as `keyGet2Custom` will return `123` for `r.path = /data/123` and `p.path = /data/:id` which is domain in `g`
I want to know which policy did the enforcer return true from. Here it is `p, role:owner, group:api, /data/:id, put`.
Like `parameters or context` in `privateEnforcer` which has the policies which is evaluated and sent in `enforce` result.
So, is there any way to know about this? 

Plus the implicit function doesn't work with my model and requirement as there are multiple grouping, so we have used custom function for that as well. It is not what I need in this particular case either. If I do get exact domain `123` then this function is used to get permission.
